### PR TITLE
QH - DSPDC-1118 - owner_primary_residence_census_division transformations

### PR DESCRIPTION
**Pull request by quazi-broad**
Updated hles_owner and hles_dog tables so that oc_primary_residence_census_division is an integer. Added transformation logic to parse the Division from the full provided string. Updated the unit test outputs to match this new logic.

### CODE REVIEW

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/OwnerTransformations.scala [hunk 1]
```diff
@@ -16,6 +16,28 @@ object OwnerTransformations {
...
+    // format is: "Division <N>: <some-description>"
+    val censusDivisionPattern: Regex = "(Division \\d+:)"
+    
+    /**
+      * Extract the Division value from the full text string.
+      *
+      * @param divisionString the text string to be parsed
+      * @return an integer of the division number
+    **/
+    def getCensusDivision(divisionString: String): (integer) = {
```
Comment (danxmoran) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/OwnerTransformations.scala:30]: Invalid type annotation here. Once you fix it the body will need tweaking to return the right type (it's currently a string)

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/OwnerTransformations.scala [hunk 2]
```diff
@@ -44,7 +66,12 @@ object OwnerTransformations {
       ocHouseholdChildCount = rawRecord.getOptionalNumber("oc_children_household"),
       ssHouseholdDogCount = rawRecord.getOptionalNumber("ss_num_dogs_hh"),
       ocPrimaryResidenceState = rawRecord.getOptional("oc_address1_state"),
-      ocPrimaryResidenceCensusDivision = rawRecord.getOptional("oc_address1_division"),
+      ocPrimaryResidenceCensusDivision = getCensusDivision(rawRecord.getOptional("oc_address1_division")),
+
+            ocSecondaryResidenceState = secondaryAddress.flatMap {
```
Comment (danxmoran) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/OwnerTransformations.scala:71]: This is already defined below, accidental copy-paste?

File: hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/OwnerTransformationsSpec.scala [hunk 3]
```diff
@@ -59,7 +59,7 @@ class OwnerTransformationsSpec extends AnyFlatSpec with Matchers {
     output.ssHouseholdDogCount shouldBe Some(2)
     // residence fields
     output.ocPrimaryResidenceState shouldBe Some("OH")
-    output.ocPrimaryResidenceCensusDivision shouldBe Some("Division 3: East North Central")
+    output.ocPrimaryResidenceCensusDivision shouldBe Some(3)
```
Comment (danxmoran) [inline hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/OwnerTransformationsSpec.scala:62]: Should this be 33? If we're extracting 3, we're losing the 2nd digit.

It looks like the other test is checking the double-digit case, so I'd actually say we should revert the source data in this one to be 3.

Comment (raaidbroad) [reply hle-survey/transformation/src/test/scala/org/broadinstitute/monster/dap/OwnerTransformationsSpec.scala:62]: Seconded; looks like it should be 33

**Pull request merged**